tests/test_monitor.py

```diff
@@ -277,6 +278,18 @@ def assert_measurement(
         )
         pynvml_mock.nvmlDeviceGetTotalEnergyConsumption.reset_mock()
 
+    def zero_energy_side_effect(handle):
+
+        energy_counters = {
+            f"handle{i}": itertools.count(start=1000, step=3)
+            for i in nvml_gpu_indices
+            if not is_old_nvml[i]
+        }
+
+        if "window0" in monitor.measurement_states:
+            return 0.0
+        return next(energy_counters[handle])
```

Comment:
I don't know why this should be implemented in such a complicated way. I was imagining something like this:

```python
pynvml_mock.nvmlDeviceGetTotalEnergyConsumption.side_effect = lambda: return 0.0
```